app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -1,19 +1,22 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   def all
-    user = User.from_omniauth(request.env['omniauth.auth'])
-    user.mailing_list = 1 if user.email
-    if user.persisted?
+    set_user
+    if @user.persisted?
       flash.notice = 'Signed in!'
-      # Uncomment below to see all of the paramaters passed back from
-      # the oauth provider:
-      # raise request.env['omniauth.auth'].to_yaml
-      sign_in_and_redirect user
+      sign_in_and_redirect @user
     else
-      session['devise.user_attributes'] = user.attributes
+      session['devise.user_attributes'] = @user.attributes
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.